hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/EthereumTransactionHandler.java

```
@@ -0,0 +1,155 @@
+
+    /**
+     * Attempts to extract the contract ID from the ethereumTransaction.
+     *
+     * @param recordItem to check
+     * @return The contract ID associated with this ethereum transaction call
+     */
+    @Override
+    public EntityId getEntity(RecordItem recordItem) {
+        var transactionRecord = recordItem.getRecord();
+
+        // pull entity from ContractResult
+        var contractFunctionResult = transactionRecord.hasContractCreateResult() ?
+                transactionRecord.getContractCreateResult() : transactionRecord.getContractCallResult();
+
+        EntityId contractId = null;
+        EntityId senderId = null;
+        EthereumTransaction ethereumTransaction = null;
+        if (!contractFunctionResult.equals(ContractFunctionResult.getDefaultInstance())) {
+            contractId = entityIdService.lookup(contractFunctionResult.getContractID());
+            senderId = EntityId.of(contractFunctionResult.getSenderId());
+
+            // no contractResult signifies a non executed transaction
+            ethereumTransaction = insertEthereumTransaction(recordItem, senderId);
+        }
+
+        if (transactionRecord.hasContractCreateResult()) {
+            insertContract(contractId, senderId, ethereumTransaction, recordItem.getConsensusTimestamp());
+        }
+
+        return contractId;
+    }
+
+    @Override
+    public TransactionType getType() {
+        return TransactionType.ETHEREUMTRANSACTION;
+    }
+
+    private EthereumTransaction insertEthereumTransaction(RecordItem recordItem, EntityId senderId) {
+        if (!entityProperties.getPersist().isEthereumTransactions()) {
+            return null;
+        }
+
+        var body = recordItem.getTransactionBody().getEthereumTransaction();
+        var ethereumDataBytes = DomainUtils.toBytes(body.getEthereumData());
+        var ethereumTransaction = ethereumTransactionParser.parse(ethereumDataBytes);
+        if (ethereumTransaction == null) {
+            return null;
```

This means that if services adds support for additional ethereum transaction types we will ignore them instead of fail. What does services do if an unsupported ethereum type is sent? Success or error code?

---

They fail. SO we should fail too instead of ignoring.
When the mirror node has replay features we can do things like this

---

If services fails, then I suggest we throw exceptions inside the parser and that way parse can never return null and this check goes away.

---

Added throw